### PR TITLE
Add support for arm64 install on GCP

### PR DESCRIPTION
**Pull request by lwan-wanglin**
Add support for arm64 install on GCP

### CODE REVIEW

**Comment by deepsm007** [comment]
@lwan-wanglin you would need to define a config to test the changes for arm64 and also run atleast 1 rehearsal for amd64 to make sure your changes doesn’t affect other jobs (eg. /pj-rehearse {test-name})

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-ipi-ovn-ipsec-p1-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-upi-baselinecaps-v411-p2-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-upi-xpn-p2-f14

**Comment by lwan-wanglin** [comment]
wait for PR https://github.com/openshift/installer/pull/7258 merge to test arm install

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-ipi-ovn-ipsec-p1-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-gcp-upi-baselinecaps-v413-p2-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-gcp-upi-baselinecaps-v413-p2-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-gcp-upi-baselinecaps-v413-p2-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-gcp-upi-baselinecaps-v413-p2-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-upi-baselinecaps-v413-p2-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-upi-baselinecaps-v413-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-gcp-upi-baselinecaps-v413-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-upi-private-xpn-sdn-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-upi-private-xpn-sdn-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-gcp-upi-xpn-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-gcp-upi-xpn-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-upi-private-xpn-sdn-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-gcp-ipi-ovn-ipsec-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-gcp-upi-xpn-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-upi-xpn-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-upi-xpn-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-gcp-upi-xpn-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-gcp-upi-xpn-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-upi-xpn-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-ipi-ovn-ipsec-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-multi-nightly-gcp-ipi-ovn-ipsec-f14

**Comment by lwan-wanglin** [comment]
- amd64 jobs installation passed.

- multi-arch installation passed, job failed becasue of one automation issue https://github.com/openshift/release/pull/41658

- arm64 jobs failed becasue this installer pr https://github.com/openshift/installer/pull/7317 not merged(being merge)

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly.yaml [hunk 1]
```diff
@@ -1855,6 +1855,36 @@ tests:
     test:
     - chain: openshift-e2e-test-qe
     workflow: baremetal-lab-agent-install-network-static
+- as: gcp-ipi-ovn-ipsec-f14
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly.yaml:1858]: I suggest to copy the same install workflow from amd nightly yaml file, and have some necessary setting for them.

Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly.yaml:1858]: So we need to set both arm and amd ci jobs to f28, right?

Comment (lwan-wanglin) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly.yaml:1858]: ah, right, will fix.

File: ci-operator/step-registry/upi/install/gcp/upi-install-gcp-commands.sh [hunk 2]
```diff
@@ -86,9 +86,13 @@ CLUSTER_NAME="$(jq -r .clusterName metadata.json)"
...
 REGION="$(jq -r .gcp.region metadata.json)"
-ZONE_0="$(gcloud compute regions describe "${REGION}" --format=json | jq -r .zones[0] | cut -d "/" -f9)"
-ZONE_1="$(gcloud compute regions describe "${REGION}" --format=json | jq -r .zones[1] | cut -d "/" -f9)"
-ZONE_2="$(gcloud compute regions describe "${REGION}" --format=json | jq -r .zones[2] | cut -d "/" -f9)"
+## Available zones and instance zones might be different in region for arm64 machines
+mapfile -t AVAILABILITY_ZONES < <(gcloud compute regions describe "${REGION}" --format=json | jq -r '.zones[]' | cut -d "/" -f9)
+mapfile -t MASTER_INSTANCE_ZONES < <(gcloud compute machine-types list --filter="zone:(${REGION}) AND name=(${CONTROL_PLANE_NODE_TYPE})" --format=json | jq -r '.[].zone')
+mapfile -t MASTER_ZONES < <(echo "${AVAILABILITY_ZONES[@]}" "${MASTER_INSTANCE_ZONES[@]}" | sed 's/ /\n/g' | sort -R | uniq -d)
+for index in {0..2}; do
+        eval ZONE_${index}=${MASTER_ZONES[index]}
```
Comment (lwan-wanglin) [inline ci-operator/step-registry/upi/install/gcp/upi-install-gcp-commands.sh:94]: will fix this space too

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-gcp-ipi-ovn-ipsec-f28

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-gcp-ipi-ovn-ipsec-f28

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-gcp-upi-xpn-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-gcp-upi-xpn-f28

File: ci-operator/step-registry/ipi/conf/gcp/ipi-conf-gcp-commands.sh [hunk 3]
```diff
@@ -19,13 +19,19 @@ fi
 
 # Do not change the default family type without consulting with cloud financial operations as their may
 # be active savings plans targeting this machine class.
-master_type=e2-standard-4
+master_type=""
+master_type_suffix=-standard-4
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/gcp/ipi-conf-gcp-commands.sh:23]: suggest to use double quote - `"-standard-4"`

File: ci-operator/step-registry/ipi/conf/gcp/ipi-conf-gcp-commands.sh [hunk 4]
```diff
@@ -19,13 +19,19 @@ fi
...
+  master_type_suffix=-standard-32
 elif [[ "${SIZE_VARIANT}" == "large" ]]; then
-  master_type=e2-standard-16
+  master_type_suffix=-standard-16
 elif [[ "${SIZE_VARIANT}" == "compact" ]]; then
-  master_type=e2-standard-8
+  master_type_suffix=-standard-8
+fi
+if [ "${OCP_ARCH}" = "amd64" ]; then
+  master_type=e2${master_type_suffix}
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/gcp/ipi-conf-gcp-commands.sh:32]: suggest to use double quote - `"e2${master_type_suffix}"`

Comment (lwan-wanglin) [reply ci-operator/step-registry/ipi/conf/gcp/ipi-conf-gcp-commands.sh:32]: the installer pr has merged, and we can install arm cluster on gcp now.  wait for your whole comments, and will fix them together

**Comment by lwan-wanglin** [comment]
@jianlinliu fixed, PTAL!

all below installation part are passed:

rehearse-40927-periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-gcp-upi-xpn-f28
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/40927/rehearse-40927-periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-gcp-upi-xpn-f28/1684734149211983872

rehearse-40927-periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-gcp-ipi-ovn-ipsec-f28
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/40927/rehearse-40927-periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-gcp-ipi-ovn-ipsec-f28/1684732051221123072

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-ipi-ovn-ipsec-p1-f14

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-upi-baselinecaps-v413-f14

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-ipi-ovn-ipsec-f28

**Comment by lwan-wanglin** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-upi-xpn-f28

**Comment by lwan-wanglin** [comment]
> /pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-upi-baselinecaps-v413-f14

this one seems to have product bug https://reportportal-openshift.apps.ocp-c1.prod.psi.redhat.com/ui/#prow/launches/-1/410753

**Comment by lwan-wanglin** [comment]
the two amd64 jobs installatoon part passed.

- rehearse-40927-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-ipi-ovn-ipsec-f28

https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/40927/rehearse-40927-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-ipi-ovn-ipsec-f28/1684778689893502976

- rehearse-40927-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-upi-xpn-f28

https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/40927/rehearse-40927-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-gcp-upi-xpn-f28/1684778771720179712

**Comment by lwan-wanglin** [comment]
@liangxia could you help to review? thanks

**Pull request merged**